src/pybind/mgr/k8sevents/module.py

```diff
@@ -80,6 +81,7 @@ def local_involved_object(self, involved_object):
     V1Event.involved_object = V1Event.involved_object.setter(local_involved_object)
 
 log = logging.getLogger(__name__)
+socket.setdefaulttimeout = 1
```

Comment:
This is actually a *function*:
```
In [1]: import socket

In [2]: help(socket.setdefaulttimeout)
Docstring:
setdefaulttimeout(timeout)

Set the default timeout in seconds (float) for new socket objects.
A value of None indicates that new socket objects have no timeout.
When the socket module is first imported, the default is None.
Type:      builtin_function_or_method
```

And more importantly, according to [pep-0554/interpreter-isolation](https://www.python.org/dev/peps/pep-0554/#interpreter-isolation), "static module data" is *shared between sub-interpreters* and AFAICS, 

https://github.com/python/cpython/blob/954900a3f98a8c0dea14dd575490237f3f8626b3/Modules/socketmodule.c#L1002

is a static module data. Meaning you cannot set this without affecting other mgr modues. 

Finally, as the Kubernetes API also makes use of `socket`, you're setting this for *Kubernetes API requests, too*.

What about just removing this line?

Reply:
`socket.setdefaulttimeout = 1` doesn't do any harm right now.